src/definitions.jl

```diff
@@ -446,6 +456,56 @@ end
 Model related functionality
 ---------------------------
 =#
+
+# Function types for the moduli with free params
+const FWScaFree = FunctionWrapper{RheoFloat,Tuple{RheoFloat,Vector{RheoFloat}}}
+const FWVecFree = FunctionWrapper{Vector{RheoFloat},Tuple{Vector{RheoFloat},Vector{RheoFloat}}}
+# Function types for the moduli with fixed params
+const FWScaFixed = FunctionWrapper{RheoFloat,Tuple{RheoFloat}}
+const FWVecFixed = FunctionWrapper{Vector{RheoFloat},Tuple{Vector{RheoFloat}}}
+# Function types for the constraint function
+const FWConstraint = FunctionWrapper{Bool,Tuple{Vector{RheoFloat}}}
+
+
+
+# constant NaN functions for default parameters and testing existence of moduli functions
+const fwscafreeNaN = ((t,p)->NaN) |> FWScaFree
+const fwvecfreeNaN = ((t,p)->fill(RheoFloat(NaN), length(t))) |> FWVecFree
+const fwscafixedNaN = (t->NaN) |> FWScaFixed
+const fwvecfixedNaN = (t->fill(RheoFloat(NaN), length(t))) |> FWVecFixed
+
+
+
+struct _RheoModel{TSca,TVec}
+
+	name::String
+	freeparams::Tuple{Vararg{Symbol, N}} where N
+	fixedparams::NamedTuple
+
+	# Moduli functions specialised for scalar and array values
+	_G::TSca
+	_Ga::TVec
+	_J::TSca
+	_Ja::TVec
+	_Gp::TSca
+	_Gpa::TVec
+	_Gpp::TSca
+	_Gppa::TVec
+
+	_constraint::FWConstraint
+	_Gramp::Bool
```

Comment:
I still don't understand the ramp nomenclature, might be worth having a longer discussion about it. Also, do we need a Jramp? Assuming it's a flag for handling singularities I'm not sure that creep moduli have any - happy to be corrected.

Reply:
You are right about Jramp, it is not required for the creep functions indeed. 
The flags is just a placeholder for future use at the moment. Nothing implemented. So yes, the long discussion is still pending!